### ISSUE TITLE
sphinx 1.6.1 introduces an TypeError error

### ISSUE THREAD

**Issue by ralonsoh** [sphinx-doc/sphinx#3765]
Subject: sphinx 1.6.1 introduces an TypeError error

### Problem
Sphinx is throwing the following error: 
2017-05-18 08:20:23.749452 |   File "/home/jenkins/workspace/gate-os-traits-docs-ubuntu-xenial/.tox/venv/local/lib/python2.7/site-packages/sphinx/themes/basic/../basic/layout.html", line 108, in template
2017-05-18 08:20:23.749476 |     {%- for css in css_files %}
2017-05-18 08:20:23.749492 | TypeError: 'NoneType' object is not iterable

This error is not happening when using version 1.5.6

#### Procedure to reproduce the problem
Execute the tox testing in os-traits
$ tox -edocs

#### Error logs / results
Sphinx version: 1.6.1
Python version: 2.7.12 (CPython)
Docutils version: 0.13.1 release
Jinja2 version: 2.9.6
Last messages:

looking for now-outdated files...
  none found
  pickling environment...
  done
  checking consistency...
  done
  preparing documents...
  done
  writing output... [ 20%] contributing
Loaded extensions:
  alabaster (0.7.10) from /opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/alabaster/__init__.pyc
  oslosphinx (unknown version) from /opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/oslosphinx/__init__.pyc
  sphinx.ext.autodoc (1.6.1) from /opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/ext/autodoc.pyc
Traceback (most recent call last):
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/setup_command.py", line 192, in run
    app.build(force_all=self.all_files)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/application.py", line 332, in build
    self.builder.build_all()
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 285, in build_all
    self.build(None, summary='all source files', method='all')
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 394, in build
    self.write(docnames, list(updated_docnames), method)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 431, in write
    self._write_serial(sorted(docnames))
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 440, in _write_serial
    self.write_doc(docname, doctree)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/builders/html.py", line 556, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/builders/html.py", line 940, in handle_page
    output = self.templates.render(templatename, ctx)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/jinja2glue.py", line 176, in render
    return self.environment.get_template(template).render(context)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/themes/basic/page.html", line 10, in top-level template code
    {%- extends "layout.html" %}
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/oslosphinx/theme/openstack/layout.html", line 2, in top-level template code
    {% set css_files = css_files + ['_static/tweaks.css'] %}
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/themes/basic/../basic/layout.html", line 128, in top-level template code
    {%- block csss %}
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/themes/basic/../basic/layout.html", line 129, in block "csss"
    {{ css() }}
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/jinja2/sandbox.py", line 427, in call
    return __context.call(__obj, *args, **kwargs)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/jinja2/runtime.py", line 553, in _invoke
    rv = self._func(*arguments)
  File "/opt/stack/os-traits/.tox/docs/local/lib/python2.7/site-packages/sphinx/themes/basic/../basic/layout.html", line 108, in template
    {%- for css in css_files %}
TypeError: 'NoneType' object is not iterable

Link: http://logs.openstack.org/61/465861/1/check/gate-os-traits-docs-ubuntu-xenial/71d6fc6/console.html

### Environment info
- OS: Ubuntu 16.04
- Python version: 2.7.12
- Sphinx version: 1.6.1

**Comment by cdeil** [comment]
Since today, I also see a TypeError from Sphinx (a different one than @ralonsoh ):
https://travis-ci.org/gammapy/gamma-cat/builds/233628422#L394
```
?25hCollecting sphinxcontrib.datatemplates
  Downloading sphinxcontrib.datatemplates-0.1.0-py2.py3-none-any.whl
Collecting gammapy from git+https://github.com/gammapy/gammapy.git#egg=gammapy
  Cloning https://github.com/gammapy/gammapy.git to /tmp/pip-build-vhzio0ha/gammapy
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-vhzio0ha/gammapy/setup.py", line 61, in <module>
        cmdclassd = register_commands(PACKAGENAME, VERSION, RELEASE)
      File "/tmp/pip-build-vhzio0ha/gammapy/astropy_helpers/astropy_helpers/setup_helpers.py", line 142, in register_commands
        from .commands.build_sphinx import (AstropyBuildSphinx,
      File "/tmp/pip-build-vhzio0ha/gammapy/astropy_helpers/astropy_helpers/commands/build_sphinx.py", line 17, in <module>
        from sphinx.setup_command import BuildDoc as SphinxBuildDoc
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/sphinx/setup_command.py", line 23, in <module>
        from sphinx.application import Sphinx
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/sphinx/application.py", line 33, in <module>
        from sphinx.environment import BuildEnvironment
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 45, in <module>
        from sphinx.util.websupport import is_commentable
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/sphinx/util/websupport.py", line 11, in <module>
        from sphinxcontrib.websupport.utils import is_commentable  # NOQA
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/sphinxcontrib/websupport/__init__.py", line 15, in <module>
        from sphinxcontrib.websupport.core import WebSupport  # NOQA
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/sphinxcontrib/websupport/core.py", line 17, in <module>
        from jinja2 import Environment, FileSystemLoader
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/jinja2/__init__.py", line 33, in <module>
        from jinja2.environment import Environment, Template
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/jinja2/environment.py", line 15, in <module>
        from jinja2 import nodes
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/jinja2/nodes.py", line 19, in <module>
        from jinja2.utils import Markup
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/site-packages/jinja2/utils.py", line 486, in <module>
        MutableMapping.register(LRUCache)
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/abc.py", line 158, in register
        if issubclass(subclass, cls):
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/abc.py", line 226, in __subclasscheck__
        if issubclass(subclass, scls):
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/abc.py", line 207, in __subclasscheck__
        ok = cls.__subclasshook__(subclass)
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/typing.py", line 860, in __extrahook__
        if issubclass(subclass, scls):
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/abc.py", line 207, in __subclasscheck__
        ok = cls.__subclasshook__(subclass)
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/typing.py", line 860, in __extrahook__
        if issubclass(subclass, scls):
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/abc.py", line 207, in __subclasscheck__
        ok = cls.__subclasshook__(subclass)
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/typing.py", line 851, in __extrahook__
        _valid_for_check(cls)
      File "/home/travis/miniconda/envs/gamma-cat/lib/python3.5/typing.py", line 837, in _valid_for_check
        raise TypeError("Parameterized generics cannot be used with class "
    TypeError: Parameterized generics cannot be used with class or instance checks
    
    ----------------------------------------
[31mCommand "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-vhzio0ha/gammapy/[0m
```
I don't think anything changed on our side, so I presume that error appeared because the build switched to Sphinx 1.6.1.

Let me know if you think this is not a Sphinx issue, or if it's a different issue and I should file a separate ticket.

**Comment by tk0miya** [comment]
Sorry for inconvenience.
The first TypeError was already fixed at #3754.

@cdeil Could you file that as another issue?

**Comment by cdeil** [comment]
@tk0miya - OK, I've opened a new issue for my TypeError here: #3766
